src/style/calc_dimension.rs

```diff
@@ -0,0 +1,58 @@
+//! Dimensions that are have to be calculated while resolving.
+
+use crate::resolve::MaybeResolve;
+
+use super::Dimension;
+
+/// An addition, subtraction, multiplication or division between [`Dimension`]s.
+///
+/// The values are calulated when the point values are resolved.
+///
+/// This type is needed, because not all calculations can be determined before resolving the actual values.
+/// For example, when adding [`Dimension::Points`] and [`Dimension::Percent`] together,
+/// the percentage first needs to be resolved to an absolute value to get the final result.
+#[derive(Clone, PartialEq, Debug)]
+pub enum CalcDimension {
+    /// Add two [`Dimension`]s together.
+    Add(Box<Dimension>, Box<Dimension>),
+
+    /// Subtract the right [`Dimension`] from the left one.
+    Sub(Box<Dimension>, Box<Dimension>),
+
+    /// Multiply a [`Dimension`] with a constant.
+    Mul(Box<Dimension>, f32),
+
+    /// Divide a [`Dimension`] by a constant.
+    Div(Box<Dimension>, f32),
```

Comment:
Why do `Mul` and `Div` not take `Box<Dimension>`?

Reply:
I tried to align with the CSS `calc()` here, which requires one side to be a `<number>`: <https://developer.mozilla.org/en-US/docs/Web/CSS/calc#syntax>

Reply:
Are there any use-cases for multiplying a `Dimension::Points` with `Dimension::Percentage`, for example?

I think multiplying with a simple factor is all you really need and then easier to write.

If there are use-cases I can change it though :)

Reply:
Thinking about it, I don’t think there are any use cases for a non constant factor here. I’m CSS I can imagine you might sometimes want to do something like `20px * (2 + 8)`, but with Taffy this can easily be handled before passing the resultant integer in.

Although for that matter I find it hard to imagine a real use case for div and mul at all. Pixels and percentages can both easily be pre-scaled. I guess maybe you might want to scale the result of an  addition or subtraction